src/openms/include/OpenMS/FORMAT/HANDLERS/MzMLHandler.h

```diff
@@ -574,42 +649,12 @@ namespace OpenMS
             tmp.setMZ(mz);
             spectrum.push_back(tmp);
 
-            //add meta data
-            UInt meta_float_array_index = 0;
-            UInt meta_int_array_index = 0;
-            UInt meta_string_array_index = 0;
-            for (Size i = 0; i < input_data.size(); i++) //loop over all binary data arrays
+            // Only if there are more than 2 data arrays, we need to check
```

Comment:
Well.then mzml would have to be changed. Also, ion mobility acquires
spectra with m/z vs intensity, they just acquire many of them...

sent from my smartphone -- you may keep any typos you find.
On Apr 14, 2015 6:12 PM, "Timo Sachsenberg" <notifications@github.com>
wrote:

> In src/openms/include/OpenMS/FORMAT/HANDLERS/MzMLHandler.h
> <https://github.com/OpenMS/OpenMS/pull/1295#discussion_r28342164>:
>
> > @@ -574,42 +649,12 @@ namespace OpenMS
> >              tmp.setMZ(mz);
> >              spectrum.push_back(tmp);
> >
> > -            //add meta data
> > -            UInt meta_float_array_index = 0;
> > -            UInt meta_int_array_index = 0;
> > -            UInt meta_string_array_index = 0;
> > -            for (Size i = 0; i < input_data.size(); i++) //loop over all binary data arrays
> > +            // Only if there are more than 2 data arrays, we need to check
>
> ok. I was just wondering if there might be some extensions like ion
> mobility or similar
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OpenMS/OpenMS/pull/1295/files#r28342164>.
>